frontend/src/component/project/Project/ProjectSettings/ProjectActions/ProjectActionsTable/ProjectActionsModal/ProjectActionsForm/FilterItem.tsx

```diff
@@ -0,0 +1,93 @@
+import { Badge, IconButton, Tooltip, styled } from '@mui/material';
+import { ConditionallyRender } from 'component/common/ConditionallyRender/ConditionallyRender';
+import { IActionFilter } from './useProjectActionsForm';
+import { Fragment } from 'react';
+import { Delete } from '@mui/icons-material';
+import Input from 'component/common/Input/Input';
+import {
+    BoxSeparator,
+    InnerBoxHeader,
+    Row,
+    StyledInnerBox,
+} from './InnerContainerBox';
+
+const StyledInput = styled(Input)(() => ({
+    width: '100%',
+}));
+
+const FilterOperation = styled(Badge)(({ theme }) => ({
+    color: 'primary',
+    margin: 'auto',
+    marginBottom: theme.spacing(1.5),
+}));
+
+export const Filter = ({
+    filter,
+    index,
+    stateChanged,
+    onDelete,
+}: {
+    filter: IActionFilter;
+    index: number;
+    stateChanged: (updatedFilter: IActionFilter) => void;
+    onDelete: () => void;
+}) => {
+    const { id, parameter, value } = filter;
+    return (
+        <Fragment key={id}>
+            <ConditionallyRender
+                condition={index > 0}
+                show={<BoxSeparator text='AND' />}
+            />
+            <StyledInnerBox>
+                <Row>
+                    <span>Filter {index + 1}</span>
+                    <InnerBoxHeader>
+                        <Tooltip title='Delete filter' arrow>
+                            <IconButton type='button' onClick={onDelete}>
+                                <Delete />
+                            </IconButton>
+                        </Tooltip>
+                    </InnerBoxHeader>
+                </Row>
+                <Row>
+                    <StyledInput
+                        label='Parameter'
+                        value={parameter}
+                        onChange={(e) =>
+                            stateChanged({
+                                id,
+                                parameter: e.target.value,
+                                value,
+                            })
+                        }
+                        error={parameter.length === 0}
+                        errorText={
+                            parameter.length === 0
+                                ? "Parameter name can't be empty"
+                                : ''
+                        }
```

Comment:
These validations are a bit too eager. I would actually say we can remove them for now and maybe filter out empty ones when we do the payload mapping.